src/python/CMSMonitoring/StompAMQ7.py

```diff
@@ -0,0 +1,497 @@
+
+    def on_message(self, frame):
+        """print debug message on_message"""
+        self.logger.debug(f"on_message HEADERS: {str(self.safe_headers(frame.headers))}, BODY: {str(frame.body)} ...")
+
+    def on_heartbeat(self):
+        """print debug message on_heartbeat"""
+        self.logger.debug('on_heartbeat')
+
+    def on_send(self, frame):
+        """print debug message on_send"""
+        self.logger.debug(f"on_send HEADERS:{str(self.safe_headers(frame.headers))}, BODY: {str(frame.body)[:160]} ...")
+
+    def on_connected(self, frame):
+        """print debug message on_connected"""
+        self.logger.debug(f"on_connected HEADERS: {str(self.safe_headers(frame.headers))}, BODY: {str(frame.body)} ...")
+
+    def on_disconnected(self):
+        """print debug message on_disconnected"""
+        self.logger.debug('on_disconnected')
+
+    def on_heartbeat_timeout(self):
+        """print debug message on_heartbeat_timeout"""
+        self.logger.debug('on_heartbeat_timeout')
+
+    def on_before_message(self, frame):
+        """print debug message on_before_message"""
+        self.logger.debug(f"on_before_message HEADERS: {str(self.safe_headers(frame.headers))}, "
+                          f"BODY: {str(frame.body)} ...")
+
+        return frame
+
+
+def broker_ips(host, port):
+    """Return broker IP addresses from provide host name"""
+    addr = []
+    for item in socket.getaddrinfo(host, int(port)):
+        # each item is (family, socktype, proto, canonname, sockaddr) tuple, so we tack 4th element sockaddr
+        # for IPv4 sockaddr is (address, port)
+        # for IPv6 sockaddr is (address, port, flow info, scope id)
+        # we are interested only in address
+        addr.append(item[4][0])
+    return addr
+
+
+# noinspection GrazieInspection
+class StompAMQ7(object):
+    """
+    Class to generate and send notifications to a given Stomp broker and a given topic.
+
+    :param username: The username to connect to the broker.
+    :param password: The password to connect to the broker.
+    :param producer: The 'producer' field in the notification header
+    :param topic: The topic to be used on the broker
+    :param validation_schema: schema to use for validation (filename of a valid json file).
+        If 'None', skip any validation. Look for schema files locally, then in 'schemas/'
+        folder in CMSMonitoring package or in folder defined in 'CMSMONITORING_SCHEMAS'
+        environmental variable.
+    :param host_and_ports: The hosts and ports list of the brokers.
+        E.g.: [('cms-test-mb.cern.ch', 61313)]
+    :param cert: path to certificate file
+    :param key: path to key file
+    :param loglevel: logging level to use for validation feedback
+    :param timeout_interval: provides timeout interval to failed broker
+    :param ipv4_only: use ipv4 servers only
+    """
+
+    # Version number is to be added in header
+    _version = '0.3'
```

Comment:
should be defined on top of this module, better should not be defined explicitly and should either go to `__init__.py` or deduced from github version

Reply:
+1